### ISSUE TITLE
Some questions about retraining

### ISSUE THREAD

**Issue by WGQ123-code** [RF5/simple-speaker-embedding#1]
Hello, it's my pleasure to see your project. I've been working on that lately. 
In your project, the model has been trained and I would like to ask if it has to be retrained if it is used for another data set. 
Also, since I didn't look closely, do I just need to run "train.ipynb" for retraining?
 Please give me some guidance. 
Thank you.

**Comment by RF5** [comment]
Hi, thanks for your interest. In short:

**Does the model need to be retrained to be used for another dataset?**
No. The model can be used for inference on any audio from any dataset (given that it is 22.05kHz audio). If the speech is very different from the datasets it has been trained on (commonvoice, librispeech, vctk), then the performance of the model will _drop_ when being used on the new data, however it should still work. Re-training or fine-tuning the model on your new dataset will improve performance, but it is not required. 

**How do I retrain the model?**
This is a bit more tricky, as the model was trained with a now-unsupported version of [fastai](https://docs.fast.ai/). Basically what you need to do is:
- Install fastai v1 in your python environment
- Pre-compute the mel-spectrograms for whatever dataset you like and put them into some folder
- Run `training.ipynb`, editing the data paths in `ds_dirs` to point to the folder where you have the preprocessed mel-spectograms.

I am planning on releasing an update with a trained 16kHz model soon though (which doesn't require fastai), which should include more abstract training code which will be easier to apply to different datasets. 

I hope that helps!

**Comment by WGQ123-code** [comment]
Thank you for your reply. It's very useful to me. 

By reading your code, I found that you also use the loss function based on GE2E, which may be the most used method now. I wonder if you have studied one-hot embedding. I also want to try to use one-hot embedding to complete my task. Whether the embedding of speakers in the data set is determined. 

For example, I have three speakers, and the embedding of the first speaker is [1,0,0]. But I see that one hot also needs training. What is the training?

I look forward to your guidance.

**Comment by RF5** [comment]
My apologies, I suspect I am not understanding your question -- maybe you can try rephrase it? If you have a one-hot embedding, no training is needed as there is a direct mapping from speaker to vector (with a one-hot function). So I don't think there will be any training. For one-hot specific speaker verification models, I have not looked into it myself because it is not my area of research. 

I might be mistaken though, so please let me know if you think I have misunderstood something

**Comment by WGQ123-code** [comment]
I'm sorry I didn't state my problem clearly. Because I've just come into contact with one hot, I'm afraid I understand it wrong.

For example, the data set I want to use now includes four speakers. Now I need to get the speaker embedding of each speaker. If I use your method, I will input a piece of data and calculate a speaker embedding through the network. 
If I use one hot, the speaker embedding of the first speaker is [1,0,0,0], and the speaker embedding of the second speaker is [0,1,0,0]. In this way, the speaker embedding can be determined directly without other calculations.From your reply, you seem to think so.

I don't know if I understand that right. If not, please give me some advice. Thanks! Have a good night.

**Comment by RF5** [comment]
Thanks for clarifying it a bit for me, I think I get it a bit better now. If you only have 4 speakers, I think directly encoding them in your one-hot way is probably easiest. But if you want to capture information about speaker similarity (how close speaker 1 is to speaker 2 compared to speaker 3), then using speaker embeddings generated by a GE2E model might be better. Either way, my recommendation if you only have 4 speakers is to use direct one-hot encoding.

**Comment by WGQ123-code** [comment]
Thank you for your further guidance. I think I have a clearer understanding of it.Thanks.
